src/compiler/parser.ts

```diff
@@ -6647,8 +6671,61 @@ namespace ts {
                     }
                 }
 
+                function textsEqual(parent: EntityName, name: EntityName): boolean {
+                    while (!ts.isIdentifier(parent) || !ts.isIdentifier(name)) {
+                        if (!ts.isIdentifier(parent) && !ts.isIdentifier(name) && parent.right.text === name.right.text) {
+                            parent = parent.left;
+                            name = name.left;
+                        }
+                        else {
+                            return false;
+                        }
+                    }
+                    return parent.text === name.text;
+                }
 
-                function tryParseChildTag(parentTag: JSDocTypeLiteral): boolean {
+                function parseChildParameterOrPropertyTag(target: PropertyLikeParse.Property): JSDocTypeTag | JSDocPropertyTag | false;

```

Comment:
I think a more likely refactoring would be to pass 1 of 2 different `tryParseChildTag` functions instead of the `target` enum. But the `target` enum pattern is also used with `parseParameterOrPropertyTag`, and it's as easy to read and (probably?) as fast to run, so I think I'll keep it this way.